expression/builtin.go

```diff
@@ -527,6 +553,29 @@ type functionClassWithName interface {
 	getDisplayName() string
 }
 
+// functions that always return not null type
+var functionSetForReturnTypeAlwaysNotNull = set.StringSet{
+	ast.IsNull: {},
+	ast.NullEQ: {},
+}
+
+// functions that always return nullable type
+var functionSetForReturnTypeAlwaysNullable = set.StringSet{
+	ast.Div:    {}, // divide by zero
+	ast.IntDiv: {}, // divide by zero
+	ast.Mod:    {}, // mod by zero
+}
+
+// functions that if all the inputs are not null, then the result is not null
+// Although most of the functions should belong to this, there may be some unknown
+// bugs in TiDB that makes it not true in TiDB's implementation, so only add a few
```

Comment:
better add a todo here.

Reply:
crdb made this as a property of function definition (mainly including pushDown-flag, nullable-flag and func-type(agg/window/normal)...) maybe we can collect them together in future.

For now, it's fine, just in case of a missing set of this.

Reply:
> better add a todo here.

Done